File: lib/api/jobs.ts
import { apiClient } from "@/lib/api/client";

export type JobListing = {
  id: string;
  domain: string;
  title: string;
  company: string;
  location: string;
  link: string;
  posted_date: string | null;
  seniority: string;
  match_reason: string;
};

export type JobRecommendations = { learner_id: string; domain: string; jobs: JobListing[]; message: string };

export function getJobRecommendations(learnerId: string) {
  return apiClient<JobRecommendations>(`/api/jobs/recommendations/${encodeURIComponent(learnerId)}`);
}

File: backend/app/schemas/jobs.py
from datetime import datetime

from pydantic import BaseModel


class JobListingResponse(BaseModel):
    id: str
    domain: str
    title: str
    company: str
    location: str
    link: str
    posted_date: datetime | None
    seniority: str
    match_reason: str = ""


class JobRecommendationsResponse(BaseModel):
    learner_id: str
    domain: str
    jobs: list[JobListingResponse]
    message: str
